src/SocketW.cpp

```diff
@@ -0,0 +1,80 @@
+#include "SocketW.hpp"
+#include <sys/errno.h>
+#include <cstring>
+#include <cmath>
+
+SocketW::SocketW(int newFd) : os(), is(), fd(newFd), consumable(false)
+{
+	
+}
+
+SocketW::~SocketW()
+{
+
+}
+
+int	SocketW::Send()
+{
+	std::string tmp;
+
+	tmp = os.str();
+	os.clear();
+	os.str(std::string());
+	tmp = tmp.substr(0, MSG_MAXLEN) + MSG_SEPARATOR;
+	return (send(fd, tmp.c_str(), tmp.size(), 0));
+}
+
+bool	SocketW::Receive()
+{
+	char	buffer[4096 + 1];
+	int		rd;
+	
+	rd = recv(fd, (void *)buffer, 4096, 0);
+	
+	if (rd < 0)
+		throw ERecvError();
+	if (rd == 0)
+		throw EConnectionClosed();
+
+	buffer[rd] = '\0';
+	is << buffer;
+	consumable = (is.str().find(MSG_SEPARATOR) != std::string::npos);
+	return (consumable);
+}
+
+std::string	SocketW::Consume()
+{
+	std::string tmp;
+	size_t		pos;
+
+	if (!consumable)
+		return (std::string("")); /* bad consume, throw exception maybe ? */
+
+	tmp = is.str();
+	pos = tmp.find(MSG_SEPARATOR);
+	is.str(std::string(tmp.substr(pos + 1)));
+	is.clear();
+	tmp = tmp.substr(0, std::min(pos, (size_t)MSG_MAXLEN));
```

Comment:
We have to chose if we keep the can consume check to call before consume or the consumable check in consume function call